transactions/WFLY-14869-microprofile-lra.adoc

```diff
@@ -0,0 +1,128 @@
+
+=== Other Interested Projects
+
+* MicroProfile
+* Galleon
+* WildFly Operator
+
+=== Relevant Installation Types
+
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+* [ ] Managed domain
+* [ ] OpenShift s2i
+* [x] Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+
+The feature pack will integrate the LRA specification as the Galleon feature pack.
+The Galleon layer will be available for provisioning.
+The provisioned feature pack will provide all functionality defined in LRA specification for users in WildFly.
+
+The feature pack will be split to two Galleon layers (i.e., two separate subsystems).
+
+The first layer provides a capability of the LRA participant - bundling the Narayana client library capable
+to parse the annotations from LRA specification and work with JAX-RS and non-JAX-RS bindings.
+
+The second layer provides the capability of the LRA coordinator - a server which provides HTTP endpoints where applications may register as participants and then the coordinator
+calls them back (via HTTP calls) when LRA finishes.
+
+The feature pack will be placed amongst other MicroProfile WildFly feature packs
+under https://github.com/wildfly/wildfly/tree/main/microprofile[the WildFly GitHub repository].
+
+Each MicroProfile spec is optional to allow for exclusion during provisioning.
+
+The LRA feature packs will be dependent on the `transactions` subsystem. The LRA uses the same object store types
+as the JTA/JTS transaction uses, it requires the recovery manager thread to be running and requires transaction
+specifics like unique node-identifier per running application server (WildFly).
+
+As said before the LRA coordinator is based on the Narayana core (`transactions` subsystem within WildFly)
+the LRA records and JTA records shares the same storage and recovery processing.
+
+=== Nice-to-Have Requirements
+
+Create a quickstart on LRA on how to use LRA with WildFly for https://github.com/wildfly/quickstart
+
+LRA records could be listed under `/subsystem=transactions/log-store=log-store/lras=...`
+and the operations (e.g. a need for probing) will be consistent to what was developed for JTA transaction types
+under `/subsystem=transactions/log-store=log-store/transactions=...`. This is optional, because the LRA Coordinator REST interface already provides this information.
+
+=== Non-Requirements
+
+NONE
+
+== Test Plan
+
+1. Testing LRA functionality within WildFly including TCK and subsystem setup.
++
+The LRA functionality tests are part of the Narayana LRA codebase (https://github.com/jbosstm/narayana/tree/master/rts/lra/test/. Namely basic and crash tests in addition to TCK.
++
+2. The WildFly will be running with LRA TCK module
++
+https://github.com/wildfly/wildfly/tree/main/testsuite/integration/microprofile-tck
++
+3. Basic integration tests which tests the Galleon layer provisioning
++
+https://github.com/wildfly/wildfly/tree/main/testsuite/integration/microprofile
++
+4. OpenShift testing
```

Comment:
I am thinking whether we should remove OpenShift from this community proposal and add such different platform test details to the product feature test plan. WDYT?